static/app/utils/profiling/renderers/textRenderer.tsx

```diff
@@ -126,19 +131,49 @@ class TextRenderer {
       // Offset x by 1x the padding
       const x = frameInPhysicalSpace[0] + HALF_SIDE_PADDING;
 
-      this.context.fillText(
-        trimTextCenter(
-          frame.frame.name,
-          findRangeBinarySearch(
-            {low: 0, high: paddedRectangleWidth},
-            n => this.measureAndCacheText(frame.frame.name.substring(0, n)),
-            paddedRectangleWidth
-          )[0]
-        ),
-        x,
-        y
+      const frameName = frame.frame.name;
+      const trim = trimTextCenter(
+        frameName,
+        findRangeBinarySearch(
+          {low: 0, high: paddedRectangleWidth},
+          n => this.measureAndCacheText(frame.frame.name.substring(0, n)).width,
+          paddedRectangleWidth
+        )[0]
       );
 
+      const {text: trimText} = trim;
+
+      if (flamegraphSearch) {
+        const searchResults = flamegraphSearch.results ?? {};
+        const searchQuery = flamegraphSearch.query;
+        const frameId = getFlamegraphFrameSearchId(frame);
+        const isFrameInSearchResults = searchResults[frameId];
+
+        if (isFrameInSearchResults && searchQuery) {
```

Comment:
I would move this to the code that does the actual searching (case where we detect a regexp and use it to search) and rely on fuse indices otherwise. The two main reasons is that the matching logic will know why a search result is included in the set and because you only want this logic to run once per query, not once per draw call. Currently, if you run a query and then zoom around the chart, all of this text matching logic will try to run at 60FPS, causing interactions to slow down and potentionally drop frames whereas ideally, at draw time, we only need to check where to draw our highlighting boxes